optuna/study.py

```diff
@@ -347,6 +347,14 @@ def objective(trial):
             n_jobs:
                 The number of parallel jobs. If this argument is set to :obj:`-1`, the number is
                 set to CPU count.
+
+                .. note::
```

Comment:
Thank you for adding this important note!

How about mentioning that the [process-based parallelization](https://optuna.readthedocs.io/en/stable/tutorial/10_key_features/004_distributed.html#sphx-glr-tutorial-10-key-features-004-distributed-py) is _"officially recommended"_ so that the user can choose that way without hesitation?

Reply:
Thank you for your clear comment! I have updated this PR by following your comment.

Reply:
How about describing with a third party voice as follows? (Thanks @hvy and @Crissman for your proofreading.)

```
``n_jobs`` allows parallelization using :obj:`threading` and may suffer from Python's GIL <https://wiki.python.org/moin/GlobalInterpreterLock>`_.
It is recommended to use :ref:`process-based parallelization<distributed>` if ``func`` is CPU bound.
```

Reply:
Awesome! It is supposed to be much better than the one I wrote initially.

Reply:
Thanks @nzw0301  for incorporating the suggestion. Sorry if it appeared somewhat intrusive to the initially proposed change.